services/motion/explore/explore.go

```diff
@@ -325,36 +328,65 @@ func (ms *explore) checkForObstacles(
 	plan motionplan.Plan,
 	obstaclePollingFrequencyHz float64,
 ) {
+	ms.logger.Debug("Current Plan")
+	for i, p := range plan {
+		ms.logger.Debugf("plan[%v]: %v", i, p)
+	}
 	// Constantly check for obstacles in path at desired obstacle polling frequency
 	ticker := time.NewTicker(time.Duration(int(1000/obstaclePollingFrequencyHz)) * time.Millisecond)
 	defer ticker.Stop()
 
+	var errCounterCurrentInputs, errCounterGenerateTransientWorldState int
 	for {
 		select {
 		case <-ctx.Done():
 			return
 		case <-ticker.C:
-			currentPose := spatialmath.NewZeroPose()
+
+			currentInputs, err := kb.CurrentInputs(ctx)
+			if err != nil {
+				ms.logger.Debugf("issue occurred getting current inputs from kinematic base: %v", err)
+				if errCounterCurrentInputs > successiveErrorLimit {
+					ms.obstacleResponseChan <- moveResponse{success: false}
+					return
+				}
+				errCounterCurrentInputs++
+			} else {
+				errCounterCurrentInputs = 0
+			}
+
+			currentPose := spatialmath.NewPose(
+				r3.Vector{X: currentInputs[0].Value, Y: currentInputs[1].Value, Z: 0},
+				&spatialmath.OrientationVector{OX: 0, OY: 0, OZ: 1, Theta: currentInputs[2].Value},
+			)
 
 			// Look for new transient obstacles and add to worldState
 			worldState, err := ms.generateTransientWorldState(ctx, obstacleDetectors)
 			if err != nil {
 				ms.logger.Debugf("issue occurred generating transient worldState: %v", err)
+				if errCounterGenerateTransientWorldState > successiveErrorLimit {
+					ms.obstacleResponseChan <- moveResponse{success: false}
+					return
+				}
+				errCounterGenerateTransientWorldState++
+			} else {
+				errCounterGenerateTransientWorldState = 0
 			}
 
-			// Select remainder of plan to check
-			ms.planMutex.Lock()
-			remainingPlan := plan[ms.planStep:]
-			ms.planMutex.Unlock()
+			// TODO: Generalize this fix to work for maps with non-transient obstacles. This current implementation
+			// relies on the plan being two steps: a start position and a goal position.
+			// JIRA Ticket: https://viam.atlassian.net/browse/RSDK-5964
+			plan[0][kb.Name().ShortName()] = currentInputs
```

Comment:
@raybjork @biotinker @nfranczak Based on a conversation with Nick, this necessary change here appears to me to be due to the fact that the differential base works in absolutes, and therefore the absolute current position needs to be 'pre-pended' to the remaining plan. 

If this is indeed the case, should this be formalized? In CheckPlan you have if statements to handle the relative nature of PTGs, should this then appear as the else statement to that if?